### ISSUE TITLE
source code for manuscript

### ISSUE THREAD

**Issue by kaczmarj** [ozanciga/self-supervised-histopathology#3]
Hello, thank you for providing the pre-trained models for https://arxiv.org/abs/2011.13971. Would it be possible to also share the source code used in the paper? For example, code to define models, train models, etc.

**Comment by ozanciga** [comment]
Currently the paper is in review so this mainly hinges on the outcome of that. We plan on releasing the trainer code sometime in May regardless. However note that this is not original work (the underlying method, i.e., SimCLR), therefore if you need the code now, I recommend checking out https://github.com/Spijkervet/SimCLR , it's a good implementation of the original Tensorflow code.

**Comment by nsmdgr** [comment]
Hello,

First, thanks a lot for your work and for sharing the pertained model with the community! We are interested in extending it, but it seems that your paper does not describe the hyperparameters and necessary preprocessing steps for the evaluation datasets (downsampling, class imbalance handling, augmentations etc.). Are you planning on releasing the code to reproduce these experiments as well or do you by any chance have pointers to preexisting repos that were used? We would be mainly interested in the classification tasks. Any help would be greatly appreciated. Thank you again!

**Comment by ozanciga** [comment]
Each image is downsampled to 224x224 pixels per patch in pretraining for most datasets. For Patch Camelyon, we use 96x96 (its native resolution). For segmentation, we use 224x224 for BACH, and 512x512 for the others. For validation, we use 1e-4 as both learning rate and the weight decay (for Resnet50+), and for Resnet18 and 34 we use 1e-5 as weight decay. In terms of augmentations, we use 90 degree rotations, flips, and medium light jittering (in pytorch, corresponds to 0.4, 0.4, 0.4, 0.2) during validation. Using more augmentations are helpful for ImageNet more than our pretrained models, since it helps ImageNet initialization to avoid overfitting. There is no balancing in either pretraining or validation. We plan on releasing the code sometime next month.

Further information on parameters at the pretraining stage:

![image](https://user-images.githubusercontent.com/25689812/114556009-44b7df00-9c36-11eb-8b1e-532621adec35.png)

![image](https://user-images.githubusercontent.com/25689812/114556105-5e592680-9c36-11eb-98b1-84a13d873118.png)

![image](https://user-images.githubusercontent.com/25689812/114556364-a2e4c200-9c36-11eb-9f17-2078d31ecd80.png)

**Comment by nsmdgr** [comment]
Thank you a lot for the detailed explanation, very helpful. Looking forward to the code release!

**Comment by tmadl** [comment]
Hello, thanks a lot for the pre-trained models, great work! Would it be possible to share the training code, to facilitate adaptation to other datasets and cancers?

**Comment by Geeks-Sid** [comment]
Hi, Now that the manuscript is out, can you share the code?
